src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashMarkStatus.java

```diff
@@ -0,0 +1,18 @@
+package stashpullrequestbuilder.stashpullrequestbuilder;
+
+import hudson.model.Result;
+import hudson.model.Run;
+
+public class StashMarkStatus {
+
+     public void handleStatus(Boolean approveOnBuildSuccessful, Boolean needsWorkOnBuildFailure, String pullRequestId,
+        Result result, StashRepository repository) {
+        if(approveOnBuildSuccessful && result == Result.SUCCESS) {
+            repository.markStatus(pullRequestId, "APPROVED");
+        }
+
+        if(needsWorkOnBuildFailure && result == Result.FAILURE) {
+            repository.markStatus(pullRequestId, "NEEDS_WORK");
+        }
+    }
+}
```

Comment:
I don't think `StashMarkStatus` needs to be a separate class. I'd rather stop proliferation of classes that have no state or map 1-to-1 to the existing classes.

Reply:
Where is the best place for this handler?

Reply:
I would just put that code where it's being called. If the code gets too long, we care refactor it across more natural lines.